### PR TITLE
RWA-3744: DMN reconfig improvement

### PR DESCRIPTION
**Pull request by adityadwadasi**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-3744 

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/MarkTaskReconfigurationService.java [hunk 1]
```diff
@@ -88,7 +89,7 @@ public TaskOperationResponse performOperation(TaskOperationRequest taskOperation
 
     private boolean isReconfigurable(String caseId) {
         List<ConfigurationDmnEvaluationResponse> results = caseConfigurationProviderService
-            .evaluateConfigurationDmn(caseId, null);
+            .evaluateConfigurationDmn(caseId, emptyMap());
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/MarkTaskReconfigurationService.java:92]: maybe a unit test needs updating to cover this?  I.e. it should fail if it goes back to null.

Comment (adityadwadasi) [reply src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/MarkTaskReconfigurationService.java:92]: Hi Ravi... I added unit test to check this behaviour using captors. Could you please verify and let me know if thats okay.

Comment (raghera) [reply src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/MarkTaskReconfigurationService.java:92]: looks good.

**Review by raghera — DISMISSED** [review]

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/MarkTaskReconfigurationServiceTest.java [hunk 2]
```diff
@@ -186,6 +191,34 @@ void should_not_mark_tasks_to_reconfigure_if_task_resource_cannot_be_reconfigura
         assertEquals(0, taskResourcesMarked);
     }
 
+    @Test
+    @SuppressWarnings("unchecked")
+    void should_fail_is_reconfigurable_when_task_attributes_are_empty() {
```
Comment (BenNzewi) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/MarkTaskReconfigurationServiceTest.java:196]: is this testing failure when task attributes are empty?

Comment (adityadwadasi) [reply src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/operation/MarkTaskReconfigurationServiceTest.java:196]: Hi Ben, Thank you pointing it. :) The test should fail when the taskAttributes are null. I should rename this appropriately.

**Review by raghera — APPROVED** [review]

**Pull request merged**